src/components/Publish/Pricing/Price.tsx

```diff
@@ -1,26 +1,40 @@
 import Conversion from '@shared/Price/Conversion'
-import { useField, useFormikContext } from 'formik'
+import { Field, useField, useFormikContext } from 'formik'
 import React, { ReactElement } from 'react'
 import Input from '@shared/FormInput'
 import Error from './Error'
 import PriceUnit from '@shared/Price/PriceUnit'
 import styles from './Price.module.css'
 import { FormPublishData } from '../_types'
+import { getFieldContent } from '../_utils'
 
 export default function Price({
-  firstPrice
+  firstPrice,
+  content
 }: {
   firstPrice?: string
+  content?: any
 }): ReactElement {
   const [field, meta] = useField('pricing.price')
 
-  const { values } = useFormikContext<FormPublishData>()
+  const { values, setFieldValue } = useFormikContext<FormPublishData>()
   const { dataTokenOptions } = values.services[0]
 
   return (
     <div className={styles.price}>
       {values.pricing.type === 'free' ? (
-        <h4 className={styles.free}>Free</h4>
+        <div className={styles.free}>
+          <Input
+            {...getFieldContent('freeAgreement', content.fields)}
+            name="pricing.freeAgreement"
+            onChange={() => {
```

Comment:
not fully sure we need that, we have bad experiences in pst with trying to overwrite Formik's default behavior. Are you sure that using `name="pricing.freeAgreement"` isn't enough? Usually this connects any input element with Formik when within a certain Formik provider which is the case here. So let's double check fi we can remove that `onChange`

Reply:
Yeah. You're right. the `onChange` method is not the right approach, the name property is not enough though so I changed the approach and use the `useField` hook

Reply:
alright, got it, think these are the rules:

- when using Formik's `Field` component only `name` is required to connect with form state
- when using any other input component directly, has to be connected with spreading `field`:
   ```
   const [field] = useField('pricing.freeAgreement')
  
   <Input {...field} />
   ```

In any case couldn't find DOM differences anymore between the 2 possibilities so went back to using `Field` again as this is how we do most other input fields in publish form.